sdk/core/core-amqp/src/retry.ts

```diff
@@ -67,13 +72,29 @@ export interface RetryConfig<T> {
   /**
    * @property {number} [delayInSeconds] Amount of time to wait in seconds before making the
    * next attempt. Default: 15.
+   * When `exponentialRetry` is set to `true`, this is used to compute the exponentially increasing delays between retries.
    */
   delayInSeconds?: number;
   /**
    * @property {string} connectionHost The host "<yournamespace>.servicebus.windows.net".
    * Used to check network connectivity.
    */
   connectionHost?: string;
+  /**
+   * @property {boolean} [exponentialRetry] Flag to denote if we want to perform exponential retry and not
+   * the default, which is linear.
+   */
+  isExponentialRetry?: boolean;
```

Comment:
@bterlson Do you foresee any other kind of retry we would be supporting? If so, then we should probably have this as an enum

Reply:
Probably, given that there are 3 or more supported retry schemes in core-http (linear, exponential, logarithmic, also see Fibonacci and a few others looking around outside Azure).

Reply:
How can a single interface `RetryOptions` cater to the different kinds of inputs required for the different retry mechanisms?
In case of libraries using the http pipeline, each retry mechanism is a different "policy", so they won't have this problem.
Logged #4199 to continue this discussion

Reply:
Updated to use `enum` as agreed

> How can a single interface RetryOptions cater to the different kinds of inputs required for the different retry mechanisms?

Agree I recall touching upon this point during initial discussion and had captured it in the assumptions as well